pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -190,20 +179,23 @@ func (w *PodWebhook) Default(ctx context.Context, obj runtime.Object) error {
 		}
 
 		suspend = jobframework.QueueNameForObject(pod.Object()) != "" || w.manageJobsWithoutQueueName
+		if suspend {
+			if pod.pod.Labels == nil {
+				pod.pod.Labels = make(map[string]string)
+			}
+			pod.pod.Labels[constants.ManagedByKueueLabelKey] = constants.ManagedByKueueLabelValue
+		}
 	}
 
 	if suspend {
 		controllerutil.AddFinalizer(pod.Object(), podconstants.PodFinalizer)
-
-		if pod.pod.Labels == nil {
-			pod.pod.Labels = make(map[string]string)
-		}
-		pod.pod.Labels[constants.ManagedByKueueLabelKey] = constants.ManagedByKueueLabelValue
-
 		gate(&pod.pod)
 
 		if features.Enabled(features.TopologyAwareScheduling) {
 			if val, ok := pod.pod.Annotations[kueuealpha.PodGroupPodIndexLabelAnnotation]; ok {
+				if pod.pod.Labels == nil {
```

Comment:
This does not seem related to the issue, because it does not say about TAS. Is it a bugfix for TAS? If so, can we move to a dedicated bugfix?

Reply:
This is not a fix. This changes need because I removed https://github.com/kubernetes-sigs/kueue/pull/4808/files/42a03da0c49ed54d38f1f198f25a05ce19dd7646#diff-8653db7737ac7710352223a2af7723aa2def7cc433c73b1d151c7502d2a9af17L198-L200

Reply:
ack